bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2RpcSocket.java

```diff
@@ -291,7 +292,7 @@ public void onText(Session session, String receivedMessage) {
                 logger.debug("{}: No Rpc listener registered for device {}, skip message: {}", thingName,
                         getString(message.src), receivedMessage);
             }
-        } catch (ShellyApiException | IllegalArgumentException e) {
+        } catch (ShellyApiException | RuntimeException e) {
```

Comment:
Runtime exceptions should not be caught unless external code is called. Is that the case here, or what can cause such exceptions?

Reply:
I made it more specific. I'm chasing a WebSocket error, which ends-up with now suitable information in the log. I suppose that this is caused by an unmatched exception. I revised exception handling to catch more specific exception. In addition onNotifyStatus() and onNotifyEvent() do not need to catch the exception, because that is handled by the caller.

Reply:
@jlaur are we fine here?

Reply:
Not yet, but the comment is obsoleted by https://github.com/openhab/openhab-addons/pull/19170#discussion_r2280370423